File: outputs/2025-07-31_14-26-54-403303/PrimeChecker.py
# Unit test result: ✅ PASSED
import math

class PrimeChecker(object):

    @staticmethod
    def main(args: list[str]):
        checker = PrimeChecker()
        numbers = [2, 4, 5, 9, 13, 17, 20]
        for num in numbers:
            if checker.is_prime(num):
                print(f'{num} é primo.')
            else:
                print(f'{num} não é primo.')

    def is_prime(self, n: int) -> bool:
        if n <= 1:
            return False
        if n == 2:
            return True
        if n % 2 == 0:
            return False
        for i in range(3, int(math.sqrt(n)) + 1, 2):
            if n % i == 0:
                return False
        return True
if __name__ == '__main__':
    PrimeChecker.main([])

File: uploads/2025-07-31_14-26-54-403303/PrimeChecker.java
public class PrimeChecker {
    public static void main(String[] args) {
        PrimeChecker checker = new PrimeChecker();
        int[] numbers = {2, 4, 5, 9, 13, 17, 20};

        for (int num : numbers) {
            if (checker.isPrime(num)) {
                System.out.println(num + " é primo.");
            } else {
                System.out.println(num + " não é primo.");
            }
        }
    }

    public boolean isPrime(int n) {
        if (n <= 1) return false;
        if (n == 2) return true;
        if (n % 2 == 0) return false;

        for (int i = 3; i <= Math.sqrt(n); i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }
}
